nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-api/src/main/java/org/apache/nifi/oauth2/OAuth2AccessTokenProvider.java

```diff
@@ -27,4 +27,14 @@ public interface OAuth2AccessTokenProvider extends ControllerService {
      * @return A valid access token (refreshed automatically if needed) and additional metadata (provided by the OAuth2 access server)
      */
     AccessToken getAccessDetails();
+
+    /**
+     * @param forceAccessTokenRefresh true if the access token should be refreshed
+     *                                even if it is not expired
+     * @return A valid access token (refreshed automatically if needed) and
+     *         additional metadata (provided by the OAuth2 access server)
+     */
+    default AccessToken getAccessDetails(final boolean forceAccessTokenRefresh) {
```

Comment:
Instead of adding this new method and option, what do you think about promoting `refreshAccessDetails()` as an interface method? This would require calling both `refreshAccessDetails()` and then `getAccessDetails()`, but it would clarify that these are separate actions, the first of which could fail, allowing the caller to determine the desired behavior. The default implementation could be nothing, making it easier for third-party implementations to implement when desired, aligning with the optional usage of the new Token Refresh Strategy.

Reply:
Yeah, you're right, I think it makes it easier from a caller perspective to understand the expected usage of those methods.

Reply:
Thinking a bit more about it... `refreshAccessDetails()`, as of now, assumes that the refresh token is not null. Some services do not send a refresh token with the access token. So I'll have to adjust the logic just a bit.